source/hpack.c

```diff
@@ -312,9 +315,9 @@ struct aws_hpack_context *aws_hpack_context_new(struct aws_allocator *allocator,
     aws_hash_table_clean_up(&context->dynamic_table.reverse_lookup);
 
 reverse_lookup_failed:
-    if (context->dynamic_table.buffer) {
-        aws_mem_release(allocator, context->dynamic_table.buffer);
-    }
+    aws_mem_release(allocator, context->dynamic_table.buffer);
```

Comment:
I like the labels, leave the `if` tho. This doesn't have to be not `NULL`.

Reply:
Dohhhhhhh